discovery-provider/src/premium_content/helpers.py

```diff
@@ -26,3 +27,15 @@ def does_user_follow_artist(user_id: int, follow_user_id: int):
             .one_or_none()
         )
         return True if result else False
+
+
+def does_user_support_artist(user_id: int, supporting_user_id: int):
+    db = db_session.get_db_read_replica()
+    with db.scoped_session() as session:
+        result = (
+            session.query(UserTip)
+            .filter(UserTip.sender_user_id == user_id)
+            .filter(UserTip.receiver_user_id == supporting_user_id)
```

Comment:
nit: the amount has to be non-zero, but might be worth being explicit here and filtering to `amount >= 0`?

Product q: Any thoughts about amount thresholds for premium content? (eg. "only tippers who have tipped at least 5 $AUDIO")

Reply:
re: product q, i think sebastian also brought this up, and the response was that track payments where price is determined by artist will kind of represent thresholds for unlocking a premium track. for tip-gated, we just check whether or not user tipped artist